src/App.tsx

```diff
@@ -1,61 +1,119 @@
+import React, { useState, useEffect } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState('');
+  const [userError, setUserError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+
+  const onTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+
+    const englishRegex = /[A-Za-z]/;
+    const ukrainianRegex = /[А-ЩЬЮЯҐЄІЇа-щьюяґєії]/;
+
+    if (englishRegex.test(e.target.value)) {
+      setTitle(e.target.value.replace(/[^A-Za-z0-9\s]*$/, ""));
```

Comment:
The regular expression used here only removes non-alphanumeric characters at the end of the string. If the requirement is to remove all non-alphanumeric characters, consider using a global regex: `/[^A-Za-z0-9\s]+/g`.